### PR TITLE
Re-land fix docker build and deploy_gallery-macos

### PR DESCRIPTION
**Pull request by christopherfujino**
## Description

This originally landed in https://github.com/flutter/flutter/pull/43362 and was reverted by https://github.com/flutter/flutter/pull/43557. The additional fix I added was changing a direct `fastlane ...` invocation in `deploy_gallery.sh` to `bundle exec fastlane ...` with the env var `BUNDLE_GEMFILE` set to the path to the `Gemfile` that was installed during the Cirrus setup step (I did the same on linux, pointing to the `Gemfile` the `Dockerfile` uses to install).

## Tests

I temporarily edited the cirrus config to run the post-submit version of `deploy_gallery-macos` and it passed (the reverted it to only exercise archiving post-submit, as before).

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: dev/bots/deploy_gallery.sh [hunk 1]
```diff
@@ -84,13 +85,14 @@ elif [[ "$OS" == "darwin" ]]; then
     fi
   )
   echo "iOS Flutter Gallery built"
-  if [[ -z "$CIRRUS_PR" ]]; then
+  #if [[ -z "$CIRRUS_PR" ]]; then
```
Comment (gspencergoog) [inline dev/bots/deploy_gallery.sh]: Remove commented out code.

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

File: dev/bots/deploy_gallery.sh (hunk 1 above)
Comment (christopherfujino) [reply dev/bots/deploy_gallery.sh]: Thanks for catching this!

**Review by mklim — APPROVED** [review]
LGTM

**Pull request merged**